airflow/auth/managers/models/menu.py

```diff
@@ -0,0 +1,28 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+
+@dataclass
+class MenuItem:
```

Comment:
I wonder if this should be somewhere else, not auth manager specific, so plugins can also use it? Of course, we can move it later.

Reply:
That's a good call. I think we better move it now to avoid to have to support multiple Airflow versions in providers. Not sure where though

Reply:
Once #47492 is merged, I think somewhere in `api_fastapi` would make sense

Reply:
I moved it in `airflow/api_fastapi/common/types.py`, does that make sense?